src/arrayMethodReduce.js

```diff
@@ -5,8 +5,24 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let accumulator;
+    let startIndex;
+
+    if (arguments.length < 2) {
+      accumulator = this[0];
+      startIndex = 1;
+    } else {
+      accumulator = initialValue;
+      startIndex = 0;
+    }
```

Comment:
```suggestion
  let accumulator = initialValue;
  let startIndex = 0;

    if (arguments.length < 2) {
      accumulator = this[0];
      startIndex = 1;
    }
```
better to avoid else